### PR TITLE
Fix #282

### LINKED ISSUE #282: the set_threshold method don't throw a TypeError exception in ironpython when a argument is not a integer
In python 2.5, if the argument is string or object, the method
"set_threshold"(in module gc) will throw an TypeError Exception(TypeError: an
integer is required), if the argument is a float, set_threshold will throw a
warning(DeprecationWarning: integer argument expected, got float) , but
neither of the two exceptions would be thrown in ironpython  

Work Item Details
--------------------
**Original CodePlex Issue:**	[Issue 8522](http://ironpython.codeplex.com/workitem/8522)
**Status:**	Active
**Reason Closed:**	Unassigned
**Assigned to:**	Unassigned
**Reported on:**	Feb 27, 2007 at 7:00 AM
**Reported by:**	junfeng
**Updated on:**	Feb 22, 2013 at 2:16 AM
**Updated by:**	jdhardy

### PR DESCRIPTION
**Pull request by slide**

### CODE REVIEW

**Comment by slozier** [comment]
There are a bunch of disabled tests in `modules/system_related/gc_test.py` for this issue (search for 8522), can you reenable them and add the file to the PR?

**Comment by slide** [comment]
Good plan. I enabled them and ran into some additional issues. So, this will be a WIP until I fix those too.

File: Languages/IronPython/IronPython.Modules/gc.cs [hunk 1]
```diff
@@ -78,9 +78,22 @@ public static class PythonGC {
...
         public static void set_threshold(CodeContext/*!*/ context, params object[] args) {
+            if(args.Length == 0) {
+                throw PythonOps.TypeError("set_threshold() takes at least 1 argument (0 given)");
+            }
+
+            if(args.Length > 3) {
+                throw PythonOps.TypeError("set_threshold() takes at most 3 arguments ({0} given)", args.Length);
+            }
+
+            if(args.Any(x => x is float)) {
```
Comment (slozier) [inline Languages/IronPython/IronPython.Modules/gc.cs]: I think you need `double` here... Or maybe not if it's passing?

Comment (slide) [reply Languages/IronPython/IronPython.Modules/gc.cs]: The test for this passes

Comment (slozier) [reply Languages/IronPython/IronPython.Modules/gc.cs]: My guess is it's not displaying this particular error message and falling through to your next check.

**Pull request merged**